src/shared/molecules/ResolvedLinkEditorPopover/ResolvedLinkEditorPopover.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { ReactNode, useRef } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useHistory, useLocation, useRouteMatch } from 'react-router';
+import { useNexusContext } from '@bbp/react-nexus';
+import { Resource } from '@bbp/nexus-sdk';
+import { clsx } from 'clsx';
+import { Tag } from 'antd';
+import { match as pmatch } from 'ts-pattern';
+import { UISettingsActionTypes } from '../../store/actions/ui-settings';
+import { RootState } from '../../store/reducers';
+import {
+  TDELink,
+  InitNewVisitDataExplorerGraphView,
+  AddNewNodeDataExplorerGraphFlow,
+} from '../../store/reducers/data-explorer';
+import { TEditorPopoverResolvedData } from '../../store/reducers/ui-settings';
+import { getOrgAndProjectFromProjectId, getResourceLabel } from '../../utils';
+import { getNormalizedTypes } from '../../components/ResourceEditor';
+import useOnClickOutside from '../../hooks/useClickOutside';
+import './styles.less';
+
+type TResultPattern = Pick<TEditorPopoverResolvedData, 'open' | 'resolvedAs'>;
+
+const PopoverContainer = ({
+  children,
+  onClickOutside,
+}: {
+  children: ReactNode;
+  onClickOutside(): void;
+}) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const {
+    editorPopoverResolvedData: { top, left, resolvedAs },
+  } = useSelector((state: RootState) => state.uiSettings);
+
+  useOnClickOutside(ref, onClickOutside);
+  return (
+    <div
+      data-testId="custom-link-popover"
+      ref={ref}
+      className={clsx(
+        'custom-popover-token',
+        resolvedAs === 'error' && 'error'
+      )}
+      style={{
+        top: `${top}px`,
```

Comment:
I found out the other day that in jsx you don't need to specify the "px" as a unit because it's used as by default. So this could change to:

```
style = {{top, left}}
```